tensorflow_addons/optimizers/lamb_optimizer_test.py

```diff
@@ -0,0 +1,461 @@
+                                                       self.evaluate(var1))
+
+    @test_utils.run_in_graph_and_eager_modes(use_gpu=False)
+    def testSharing(self):
+        for dtype in [tf.dtypes.half, tf.dtypes.float32, tf.dtypes.float64]:
+            with self.cached_session():
+                # Initialize variables for numpy implementation.
+                m0, v0, m1, v1 = 0.0, 0.0, 0.0, 0.0
+                var0_np = np.array([1.0, 2.0], dtype=dtype.as_numpy_dtype)
+                grads0_np = np.array([0.1, 0.1], dtype=dtype.as_numpy_dtype)
+                var1_np = np.array([3.0, 4.0], dtype=dtype.as_numpy_dtype)
+                grads1_np = np.array([0.01, 0.01], dtype=dtype.as_numpy_dtype)
+
+                var0 = tf.Variable(var0_np)
+                var1 = tf.Variable(var1_np)
+                grads0 = tf.constant(grads0_np)
+                grads1 = tf.constant(grads1_np)
+                opt = lamb_optimizer.LAMBOptimizer()
+
+                if not tf.executing_eagerly():
+                    update1 = opt.apply_gradients(
+                        zip([grads0, grads1], [var0, var1]))
+                    update2 = opt.apply_gradients(
+                        zip([grads0, grads1], [var0, var1]))
+                    self.evaluate(tf.compat.v1.global_variables_initializer())
+
+                # Fetch params to validate initial values
+                self.assertAllClose([1.0, 2.0], self.evaluate(var0))
+                self.assertAllClose([3.0, 4.0], self.evaluate(var1))
+
+                # Run 3 steps of intertwined LAMB1 and LAMB2.
+                for t in range(3):
+                    beta_1_power, beta_2_power = get_beta_accumulators(
+                        opt, dtype)
+                    self.assertAllCloseAccordingToType(
+                        0.9**(t + 1), self.evaluate(beta_1_power))
+                    self.assertAllCloseAccordingToType(
+                        0.999**(t + 1), self.evaluate(beta_2_power))
+
+                    if not tf.executing_eagerly():
+                        if t % 2 == 0:
+                            update1.run()
+                        else:
+                            update2.run()
+                    else:
+                        opt.apply_gradients(
+                            zip([grads0, grads1], [var0, var1]))
+
+                    var0_np, m0, v0 = lamb_update_numpy(
+                        var0_np, grads0_np, t, m0, v0)
+                    var1_np, m1, v1 = lamb_update_numpy(
+                        var1_np, grads1_np, t, m1, v1)
+
+                    # Validate updated params
+                    self.assertAllCloseAccordingToType(var0_np,
+                                                       self.evaluate(var0))
+                    self.assertAllCloseAccordingToType(var1_np,
+                                                       self.evaluate(var1))
+
+    @test_utils.run_in_graph_and_eager_modes(use_gpu=False)
```

Comment:
Is there a reason for disabling `gpu`? We'll need to make sure the optimizer is compatible with a model running on GPU

Perhaps look at this disabling of half precision: https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/conditional_gradient_test.py#L145